sublime_jedi/completion.py

```diff
@@ -153,9 +153,18 @@ def on_query_completions(self, view, prefix, locations):
             self.is_completion_ready = None
 
             if self.completions:
-                cplns = [tuple(i) for i in self.completions]
+                # sort completions by frequency in document
+                buffer = view.substr(sublime.Region(0, view.size()))
+                cplns = sorted(
+                    self.completions,
+                    key=lambda x: (
+                        -buffer.count(x[1]),  # frequency in the text
```

Comment:
`buffer` is a string that has the content of the current view, meaning all the text of the current file.

I tried what you say, and here are the results:

I have an old dual core processor from 10 years ago, a L7700 at 1.8GHz.

The current implementation in the sorting part, to execute this section takes 1 ms.


```python
buffer = view.substr(sublime.Region(0, view.size()))
cplns = sorted(
    self.completions,
    key=lambda x: (
        -buffer.count(x[1]),  # frequency in the text
        len(x[1]) - len(x[1].strip('_')),  # how many undescores
        x[1]  # alphabetically
    )
)
```

This implementation using `view.find_all` takes 42ms

```python
cplns = sorted(
    self.completions,
    key=lambda x: (
        -len(view.find_all(x[1])),  # frequency in the text
        len(x[1]) - len(x[1].strip('_')),  # how many undescores
        x[1]  # alphabetically
    )
)
```

I guess beacuse searching just raw strings is easier than using regex, and because of the biding between Python and C++.